src/plugins/data/public/query/query_string/dataset_service/lib/index_type.ts

```diff
@@ -158,9 +152,22 @@ const fetchIndices = async (dataStructure: DataStructure): Promise<string[]> =>
 
   const searchResponseToArray = (response: any) => {
     const { rawResponse } = response;
-    return rawResponse.aggregations
-      ? rawResponse.aggregations.indices.buckets.map((bucket: { key: any }) => bucket.key)
-      : [];
+    if (!rawResponse.aggregations) {
+      return [];
+    }
+
+    return rawResponse.aggregations.indices.buckets.map((bucket: { key: string }) => {
+      const key = bucket.key;
+      // Handle the case of serverless cluster where key format is either:
+      // - datasource-id::TIMESERIES::<index-name>:0
+      // - datasource-id::<index-name>:0
+      // Note: Index names cannot contain ':' or '::' in OpenSearch, so these delimiters
+      // are guaranteed to be part of the serverless format, not the index name
+      const parts = key.split('::');
+      const lastPart = parts[parts.length - 1] || '';
```

Comment:
just verifying there's no way we would run into an exception here right?

Reply:
Yes. I verified both : and ::. Error is 400 invalid index name. Was trying to add screenshot but failed.